cmd/controller-manager/app/controller_manager.go

```diff
@@ -90,19 +90,20 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 	// Build the K8s kubeconfig / client / clientBuilder
 	glog.V(4).Info("Building k8s kubeconfig")
 
-	k8sKubeconfig, err := rest.InClusterConfig()
+	var err error
+	var k8sKubeconfig *rest.Config
+	if controllerManagerOptions.K8sAPIServerURL == "" && controllerManagerOptions.K8sKubeconfigPath == "" {
+		k8sKubeconfig, err = rest.InClusterConfig()
+	} else {
+		k8sKubeconfig, err = clientcmd.BuildConfigFromFlags(
+			controllerManagerOptions.K8sAPIServerURL,
+			controllerManagerOptions.K8sKubeconfigPath)
```

Comment:
That does all of the work, but doesn't give me anything I can use besides a client specific to Kubernetes. Is there a way to pull the resolved config out of there? It's been a while since I was even trying to do what I was trying to do. I'll take a look again later.